File: file_processing/pl/asc_numeric_file_sort.pl
#!/usr/bin/perl

###################################################
## slurp lines from file of numbers 
## and sort them numerically using the default
## sort.
## To run:
## >> perl asc_numeric_file_sort.pl numbers.txt
## @author: vladimir kulyukin
###################################################
use strict;
use warnings;

# 1. get the 0th command line argument
my $infile = shift;
# 2. open the file for reading
open(IN, '<', $infile) or die $!;
# 3. read lines, sort, and print
my @lines = <IN>;
print sort { $a <=> $b } @lines;

File: file_processing/py/asc_numeric_file_sort.py
#!/usr/bin/python

#######################################
# Read in a file of numbers and
# sort them from smallest to largest.
# @author: vladimir kulyukin
#######################################


import sys

# 1. get 1st command line arg
file_path = sys.argv[1]
# 2. open file_path for reading
infile = open(file_path, 'r')
# 3. read the lines, sort, print
for n in sorted([int(line) for line in infile.readlines()]):
    print n

